### DISCUSSION TITLE
Load relation from a field resolver

### DISCUSSION THREAD

**Discussion by lednhatkhanh** [prisma/prisma#1899] [Q&A] [answered]
I have 2 models:

```
model User {
  id        String   @id @default(uuid())
  email     String   @unique
  posts     Post[]
}

model Post {
  id        String   @id @default(uuid())
  content   String
  user      User
}
```

Also, I have a field:

```
Post {
 user: (parent) => User
}
```

Currently, to be able to get the user, I have to query the post again:

```typescirpt
return await prisma.post.findOne({ where: { id: parent.id } }).user();
```

So basically I have to query the post twice and then query the user, so the question is: Is there a way for me to access the relation field, in this case `Post.user` to use it for my query? Thank you.

**Comment by lednhatkhanh** [comment]
Ended up using raw query with an inner join clause:

```
`SELECT "public"."User"."id", "public"."User"."email", "public"."User"."name", "public"."User"."password", "public"."User"."createdAt", "public"."User"."updatedAt" from "public"."User" INNER JOIN "public"."Post" ON "public"."User"."id" = "public"."Post"."user" WHERE "public"."Post"."id" = '${post.id}';`,
```

I still hope that there will be a faster way to do this rather than writing a raw query since this is a common use case IMHO.

**Comment by pantharshit00** [comment, marked as answer]
Hi, @lednhatkhanh 

You can instead load the relation using the include syntax:
```ts
return await prisma.post.findOne({ where: { id: parent.id }, include:{ user: true} )
```
That would give you both user and post data.

Docs: https://github.com/prisma/prisma2/blob/master/docs/prisma-client-js/api.md#manipulating-the-selection-set

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
